pkg/controller.v1/common/pod.go

```diff
@@ -392,9 +392,9 @@ func (jc *JobController) ReconcilePods(
 			}
 			// Check if the pod is retryable.
 			if spec.RestartPolicy == apiv1.RestartPolicyExitCode {
-				if pod.Status.Phase == v1.PodFailed && trainutil.IsRetryableExitCode(exitCode) {
+				if pod.Status.Phase == v1.PodFailed && !trainutil.IsRetryableExitCode(exitCode) {
```

Comment:
I think this was intended but we just need to improve the log message

Reply:
The original logic here is to provide flexibility to define your own retryable exit code. This should not be changed. 

Since kubernetes doesn't have restart method, That's the reason we `delete` the pod, and wait for next `reconcile` loop to create a new one. It's asynchronous so `restart` here may be confused to users.